ingest/ledgerbackend/buffered_meta_pipe_reader.go

```
@@ -99,6 +99,10 @@ func (b *bufferedLedgerMetaReader) readLedgerMetaFromPipe() (*xdr.LedgerCloseMet
 		// Wait for LedgerCloseMeta buffer to be cleared to minimize memory usage.
 		select {
 		case <-b.runner.getProcessExitChan():
+			if untilSequence != 0 {
```

we also handle `<-b.runner.getProcessExitChan` a few lines below. Should we also have a `untilSequence != 0 ` check in that block of code?

---

looks like `getProcessExitChan()` is handled a few lines above as well

---

We only handle the other instances of `getProcessExitChan` if an error occurs (so it ignores cases when core is done but there are still ledgers in a buffer). However, I think there's a bug. In `for frameLength > metaPipeBufferSize && len(b.c) > 0` we should return an error if core was closed with an error. Going to add it in a new commit.

---

OK fixed in aecfd759c1.

---

do we still need to worry about the case where `len(b.c) > 0` but `frameLength <= metaPipeBufferSize` ?

---

In such case (and on Core exit) we won't enter into a `for` loop but the `xdr.Unmarshal` below will error and we will handle the error there.